src/main/java/Duke.java

```diff
@@ -1,10 +1,258 @@
+            return;
+        }
+        System.out.println(MESSAGE_TASKS_IN_LIST);
+        for (Task task : tasks) {
+            System.out.println(" " + taskNumber + "." + task);
+            taskNumber++;
+        }
+        printSeparator();
+    }
+
+    /**
+     * Sets isDone of selected task to true
+     *
+     * @params tasks  Array of current stored tasks
+     * @params splitUserInput  Array of strings after original user input has been split by whitespace
+     * @params taskCount  Current count of tasks stored
+     */
+    public static void setTaskAsDone(Task[] tasks, String[] splitUserInput, int taskCount) {
+        //Determine index of task to be marked as done - this part can pass in broken array straight (To change)
+        if (splitUserInput.length == 1) {
+            System.out.println(MESSAGE_ERROR_INVALID_COMMAND);
+            printSeparator();
+            return;
+        }
+        int taskNumber = Integer.parseInt(splitUserInput[1]) - 1;
+        printSeparator();
+
+        //Check if taskNumber is out of bounds
+        if (taskNumber < 0 || taskNumber > taskCount - 1) {
+            System.out.println(MESSAGE_ERROR_INVALID_NUMBER);
+            printSeparator();
+            return;
+        }
+
+        //TaskNumber is valid
+        tasks[taskNumber].setAsDone();
+        System.out.println(MESSAGE_TASK_MARKED);
+        System.out.println("  [" + "\u2713" + "] " + tasks[taskNumber].description);
+        printSeparator();
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String logo = "____    __    ____  ___       __      .___________. _______ .______      \n"
+                + "\\   \\  /  \\  /   / /   \\     |  |     |           ||   ____||   _  \\     \n"
+                + " \\   \\/    \\/   / /  ^  \\    |  |     `---|  |----`|  |__   |  |_)  |    \n"
+                + "  \\            / /  /_\\  \\   |  |         |  |     |   __|  |      /     \n"
+                + "   \\    /\\    / /  _____  \\  |  `----.    |  |     |  |____ |  |\\  \\----.\n"
+                + "    \\__/  \\__/ /__/     \\__\\ |_______|    |__|     |_______|| _| `._____|\n";
+        String endLogo = "________              \n"
+                + "___  __ )____  ______ \n"
+                + "__  __  |_  / / /  _ \\\n"
+                + "_  /_/ /_  /_/ //  __/\n"
+                + "/_____/ _\\__, / \\___/ \n"
+                + "        /____/        ";
+        String userInput;
+        String[] splitUserInput;
+        Task[] tasks = new Task[ARRAY_SIZE_TASKS];
+        int taskCount = 0;
+        boolean isFinished = true;
+        Scanner in = new Scanner(System.in);
```

Comment:
Should these be extracted as a separate method to initialize the variables?

Reply:
I will take a look at this again, thanks for the suggestion